src/utils.ts

```diff
@@ -170,16 +216,27 @@ function findNSPrefix(subset) {
  * @param {object} namespaceResolver - xpath namespace resolver
  * @returns {Array} i.e. [{prefix: "saml", namespaceURI: "urn:oasis:names:tc:SAML:2.0:assertion"}]
  */
-function findAncestorNs(doc, docSubsetXpath, namespaceResolver) {
+export function findAncestorNs(
+  doc: Node,
+  docSubsetXpath: string,
+  namespaceResolver?: XPathNSResolver
+) {
   const docSubset = xpath.selectWithResolver(docSubsetXpath, doc, namespaceResolver);
+  let elementSubset: Element[] = [];
 
-  if (!Array.isArray(docSubset) || docSubset.length < 1) {
+  if (!isArrayHasLength(docSubset)) {
     return [];
   }
 
+  if (!docSubset.every((node) => xpath.isElement(node))) {
+    throw new Error("Document subset must be list of elements");
+  } else {
+    elementSubset = docSubset as Element[];
+  }
```

Comment:
I think there is an opportunity to introduce something like...
```suggestion
  function isElementSubset(docSubset: Node[]): docSubset is Element[] {
    return docSubset.every((node) => xpath.isElement(node))
  }

  if (!isElementSubset(docSubset)) {
    throw new Error("Document subset must be list of elements");
  }
```

so you avoid having to use a separate `elementSubset` and coerce docSubset into another type